library/std/src/primitive_docs.rs

```diff
@@ -11,8 +11,9 @@
 /// `bool` implements various traits, such as [`BitAnd`], [`BitOr`], [`Not`], etc.,
 /// which allow us to perform boolean operations using `&`, `|` and `!`.
 ///
-/// `if` always demands a `bool` value. [`assert!`], which is an important macro in testing,
-/// checks whether an expression returns `true` and panics if it isn't.
+/// `if` always demands a `bool` value as its conditional. [`assert!`], which is an
+/// important macro in testing, checks whether an expression is `true` and panics
+/// if it isn't.
```

Comment:
Could reword 

> checks whether an expression is `true` and panics if it isn't.

as 

> panics if the expression is `false`.

Reply:
Are you suggesting

> checks whether an expression is `true` and panics if it's `false`

or something else? I feel like maybe it's good enough as it is now, but open to changing it.

Reply:
Thankfully Rust isn’t Haskell and doesn’t have to worry about pervasive bottoms in data, so it really *is* the case for us that “isn’t `true`” necessarily implies “is `false`”.  😄